scripts/pylib/twister/twisterlib/runner.py

```diff
@@ -837,6 +837,9 @@ def run(self):
                     self.defconfig['CONFIG_FAKE_ENTROPY_NATIVE_POSIX'] == 'y'):
                     instance.handler.seed = self.options.seed
 
+            if self.options.extra_test_args and instance.platform.name.startswith("native_posix"):
```

Comment:
You should not be checking for platform name here, there are native_posix configurations that do not use that name (nrf52_bsim), instead I would check for the architecture (posix).

Reply:
Changed to `... instance.platform.arch == "posix":`